core/deployment/src/main/java/io/quarkus/deployment/builditem/ApplicationInfoBuildItem.java

```diff
@@ -19,4 +34,16 @@ public String getName() {
     public String getVersion() {
         return version;
     }
+
+    public String getFinalName() {
```

Comment:
I'm not sure how I feel about promoting these three new properties; what backends would consume this information and why?

Reply:
There is a possibility that there could be more than one output from the augmentation process.  Which final name is this in that case?

Reply:
I could see promoting group in here if we agree on what it actually represents.  If I understand correctly, the Maven group ID is being assigned to this group, and this group then becomes the Kubernetes group?  If so, that's a little surprising: Kubernetes groups are a completely different concept from Maven group ID, with just a coincidence of name.

Reply:
So, the group does default to the maven groupId indeed. The idea is that you need a group in order to properly create a docker image that will be used by Kubernetes. We could of course drop group from here and just keep it in `KubernetesConfig`, but as I mentioned above it feels a little weird from a user perspective.

As for the other fields, they are needed by this extension in order to know where to write the resources and which binary will be used (as part of the yet to written openshift work for this extension).
From my point of views, these are just properties from the build environment that various extensions could take advantage of ( this one obviously does :) ).

Reply:
So maybe group ID is OK as a general information input.  I think you could also introduce a Kubernetes config property as well though, called `group` with a default value of `${quarkus.application.group}`.  This lets us document the behavior and also override it without forcing the group name of the input application to change.

Reply:
Sure thing

Reply:
Just to be on the safe side, is `${quarkus.application.group}` supposed to be expanded automatically? Because I tried it and it's **not** being expanded :)

Reply:
Removing the other properties is possible, but that would mean that we would have to forgo the ability to do a docker builds and application of the Kubernetes resources to the cluster - we would essentially be left with the resource generation only.

Reply:
> Just to be on the safe side, is `${quarkus.application.group}` supposed to be expanded automatically? Because I tried it and it's **not** being expanded :)

Yes it is supposed to be.  The expansion would be done when the configuration is read.  This actually happens in two stages though: at build time, and again at run time.  So you can't (for example) expand a run time property in a build time config item (build time happens first).

Reply:
> Removing the other properties is possible, but that would mean that we would have to forgo the ability to do a docker builds and application of the Kubernetes resources to the cluster - we would essentially be left with the resource generation only.

The problem with wiring classes and base dir is that these aren't being used to configure the build, rather they're just data carriers which carry the actual values.  This is a problem waiting to happen.  I have no problem introducing configuration properties to control these things, but they're specific to augmentation, not to the application.

That said, the information carried by these two properties *could* probably be derived in some other way.  I'll have to look at the changes some more.

I think we do (or we should) have build items representing the output target artifact(s).

Reply:
In this case both are `BUILD_TIME` - `ApplicationConfig` and `KubernetesConfig`. The default value of `${quarkus.application.name}` is not being expanded. Should I open an issue for this?

Reply:
Sure, go ahead and open an issue.  Thanks.

Reply:
No problem, here it is: https://github.com/quarkusio/quarkus/issues/1703